core/src/main/kotlin/com/justai/jaicf/logging/ConversationLogger.kt

```diff
@@ -0,0 +1,110 @@
+package com.justai.jaicf.logging
+
+import com.justai.jaicf.activator.ActivationContext
+import com.justai.jaicf.api.BotRequest
+import com.justai.jaicf.context.BotContext
+import com.justai.jaicf.context.LoggingContext
+import com.justai.jaicf.helpers.logging.WithLogger
+import com.justai.jaicf.reactions.Reaction
+
+/**
+ * Main abstraction for class, which will perform dialog logging to any service or console.
+ * It supports log obfuscation by [ConversationLogObfuscator] to hide sensitive data.
+ *
+ * @property logObfuscator implementation
+ *
+ * @see ConsoleConversationLogger
+ * */
+interface ConversationLogger {
+
+    /**
+     * Implementation of [ConversationLogObfuscator], which will be used to obfuscate sensitive data in request and reactions.
+     * */
+    val logObfuscator: ConversationLogObfuscator?
+
+    /**
+     * Produces log to console or external consumer.
+     *
+     * @param activationContext current request's [ActivationContext]
+     * @param botContext a current user's [BotContext]
+     * @param request a current user's [BotRequest]
+     * @param loggingContext current request's [LoggingContext]
+     * */
+    fun produce(
```

Comment:
Why not to pull all these arguments inside the `LoggingContext` and pass only a single logging context to this function?